string who
string what
string where
---
bool who
bool what
bool where
